### PR TITLE
Fetch klib artifacts too

### PR DESCRIPTION
**Pull request by alexarchambault**
Quick workaround for https://github.com/com-lihaoyi/mill/issues/3695

### CODE REVIEW

**Comment by alexarchambault** [comment]
The fix here (currently, second commit) should only be temporary. Working on a proper fix, that should be two-fold:
- Mill should allow users to add extra artifact types (sbt does for sure, Maven and Gradle probably do too)
- coursier should fetch `klib` artifacts by default, given that these seem pretty widespread in the Kotlin ecosystem

File: main/util/src/mill/util/CoursierSupport.scala [hunk 1]
```diff
@@ -104,7 +108,7 @@ trait CoursierSupport {
             Agg.from(
               res.files
                 .map(os.Path(_))
-                .filter(path => path.ext == "jar" && resolveFilter(path))
+                .filter(resolveFilter)
```
Comment (lefou) [inline main/util/src/mill/util/CoursierSupport.scala:111]: To preserve previous behavior, you should change the `resolveFilter` default to contain the `path.ext == "jar"` match, I guess.

**Comment by lihaoyi** [comment]
Just tried this out, seems to satisfy the manual test described in ./mill -i dist.run example/kotlinlib/web/3-hello-kotlinjs foo.compile

File: main/util/src/mill/util/CoursierSupport.scala (hunk 1 above)
Comment (alexarchambault) [reply main/util/src/mill/util/CoursierSupport.scala:111]: Answered in https://github.com/com-lihaoyi/mill/pull/3701#discussion_r1794821092

**Comment by alexarchambault** [comment]
Closing this in favor of https://github.com/com-lihaoyi/mill/pull/3701 and https://github.com/com-lihaoyi/mill/pull/3705 (the two PRs are needed to fix https://github.com/com-lihaoyi/mill/issues/3695)

**Pull request closed without merging**